orbit/pkg/useraction/mdm_migration_darwin.go

```diff
@@ -563,3 +581,223 @@ func (m *swiftDialogMDMMigrator) MigrationInProgress() (bool, error) {
 func (m *swiftDialogMDMMigrator) MarkMigrationCompleted() error {
 	return m.mrw.RemoveFile()
 }
+
+func StartMDMMigrationOfflineWatcher(ctx context.Context, client *service.DeviceClient, swiftDialogPath string, swiftDialogCh chan struct{}, fileWatcher migration.FileWatcher) {
+	if cap(swiftDialogCh) != 1 {
+		log.Fatal().Msg("swift dialog channel must have a buffer size of 1")
+	}
+
+	watcher := &offlineWatcher{
+		client:          client,
+		swiftDialogPath: swiftDialogPath,
+		swiftDialogCh:   swiftDialogCh,
+		fileWatcher:     fileWatcher,
+	}
+
+	// start loop with 3-minute interval to ping server and show dialog if offline
+	go func() {
+		ticker := time.NewTicker(1 * time.Minute)
+		defer ticker.Stop()
+
+		log.Info().Msg("starting offline dialog loop")
+		for {
+			select {
+			case <-ctx.Done():
+				log.Info().Msg("stopping offline dialog loop")
+				return
+			case <-ticker.C:
+				log.Debug().Msg("offline dialog, got tick")
+				go watcher.processTick(ctx)
+			}
+		}
+	}()
+}
+
+type offlineWatcher struct {
+	client          *service.DeviceClient
+	swiftDialogPath string
+	// swiftDialogCh is shared with the migrator and used to ensure only one dialog is open at a time
+	swiftDialogCh chan struct{}
+	fileWatcher   migration.FileWatcher
+}
+
+func (o *offlineWatcher) processTick(ctx context.Context) {
+	// try to block the dialog channel
+	select {
+	case o.swiftDialogCh <- struct{}{}:
+		log.Debug().Msg("offline dialog, blocking dialog channel")
+	default:
+		log.Debug().Msg("offline dialog, dialog channel already blocked")
+		return
+	}
```

Comment:
feels like we're using channels as if they were mutexes, any reason not to use that? otherwise no need to take the lock, you can just "queue" a message

Reply:
Maybe using "blocked" in the debug/comments is confusing. In this case, I think we're looking to explicitly drop messages that might otherwise be "queued" when the channel is full (i.e. "blocked") so that we don't have dialogs popping up one after the other. 

I did think about using a mutex, but it seems like a channel is more idiomatic. In particular, I think a mutex isn't really meant for non-blocking control. There is the `TryLock` method, but it is discouraged in the [docs](https://pkg.go.dev/sync#Mutex.TryLock):
```
TryLock tries to lock m and reports whether it succeeded.

Note that while correct uses of TryLock do exist, they are rare, and use of TryLock is often a sign of a deeper problem in a particular use of mutexes.
```